app/resolve-guid/guid-route.ts

```diff
@@ -17,10 +17,21 @@ export default abstract class GuidRoute extends Route.extend({
     getModel: task(function *(this: GuidRoute, guid: string) {
         const blocker = this.ready.getBlocker();
 
-        const model = yield this.store.findRecord(this.modelName(), guid, {
-            include: this.include(),
-            adapterOptions: this.adapterOptions(),
-        });
+        let model;
+        try {
+            model = yield this.store.findRecord(this.modelName(), guid, {
+                include: this.include(),
+                adapterOptions: this.adapterOptions(),
+            });
+        } catch (e) {
+            // To do custom error handling, add an error() action to the route that subclasses GuidRoute.
+            if (this.error) {
+                this.send('error');
+            } else {
+                this.send('error');
+                throw e;
```

Comment:
We probably want to throw in either case, so the task instance errors. Instead of the if/else:
```
} catch (e) {
  this.send('error');
  throw e;
}